src/radiation_system.hpp

```diff
@@ -426,6 +432,26 @@ void RadSystem<problem_t>::ReconstructStatesPLM(arrayconst_t &q_in, array_t &lef
 	});
 }
 
+template <typename problem_t>
+template <FluxDir DIR>
+void RadSystem<problem_t>::ReconstructStatesAVE(arrayconst_t &q_in, array_t &leftState_in, array_t &rightState_in, amrex::Box const &indexRange,
+						const int nvars)
+{
+	// construct ArrayViews for permuted indices
+	quokka::Array4View<amrex::Real const, DIR> q(q_in);
+	quokka::Array4View<amrex::Real, DIR> leftState(leftState_in);
+	quokka::Array4View<amrex::Real, DIR> rightState(rightState_in);
+
+	amrex::ParallelFor(indexRange, nvars, [=] AMREX_GPU_DEVICE(int i_in, int j_in, int k_in, int n) noexcept {
+		// permute array indices according to dir
+		auto [i, j, k] = quokka::reorderMultiIndex<DIR>(i_in, j_in, k_in);
+
+		const auto ave = 0.5 * (q(i - 1, j, k, n) + q(i, j, k, n));
```

Comment:
My suggestion was to modify the existing reconstruction so that it interpolates between PPM and this expression as a function of optical depth.

Reply:
I did interpolation between PPM and AVERAGE in Lines 909 - 946.